### PR TITLE
[Core] Adding coefficients to variable redistance

### PR DESCRIPTION
**Pull request by ddiezrod**
**📝 Description**
This closes the problem exposed in https://github.com/KratosMultiphysics/Kratos/issues/9534. A couple of comments:
- I have no idea how to name these new variables, I thought it could be less confusing to just call them "coefficients" as its difficult to assign a very clear meaning to them, but Im very open to use more descriptive names.
- Variational redistance is not very clean right now. There are several things that could be improved: divide it in cpp/header, add constructor with settings, replace local flag with an option from settings... As its quite a lot I think it might be better to fix this, as its more urgent, and do the cleaning later.

**🆕 Changelog**
- Adding option to change some "magical" coefficients in variational redistance process from constructor.

### CODE REVIEW

File: kratos/includes/variables.h [hunk 1]
```diff
@@ -470,6 +470,10 @@ namespace Kratos
     //for Geometry Variables
     KRATOS_DEFINE_3D_VARIABLE_WITH_COMPONENTS(PARAMETER_2D_COORDINATES)
 
+    // Variational redistance
+    KRATOS_DEFINE_VARIABLE(double, VARIATIONAL_REDISTANCE_COEFFICIENT_1)
```
Comment (loumalouomega) [inline kratos/includes/variables.h]: I don't like the names....

**Review by loumalouomega — COMMENTED** [review]
I don't like the names. I think it is more elegant _FIRST and _SECOND. Not blocking, just saying

File: kratos/includes/variables.h (hunk 1 above)
Comment (ddiezrod) [reply kratos/includes/variables.h]: Ok, I will rename them, if nobody has a better option

**Comment by ddiezrod** [comment]
ping @KratosMultiphysics/technical-committee

**Review by RiccardoRossi — DISMISSED** [review]
ok

**Comment by ddiezrod** [comment]
I renamed the coeffs, following @loumalouomega suggestion. Can someone reapprove?

**Review by rubenzorrilla — APPROVED** [review]

**Comment by ddiezrod** [comment]
It seems failure in nightly tests come from other place. Merging.

**Pull request merged**